### ISSUE TITLE
Use struct and Drop trait

### ISSUE THREAD

**Issue by StefanSchoof** [martindeegan/rust_pigpio#4]
Hello, I new to the rust and gpio world. I think this wrapper should implement the [Drop trait](https://doc.rust-lang.org/std/ops/trait.Drop.html). Having a simple struct with the version as property and the `gpioInitialise` called in the constructor. This would prevent errors in library for not initialized and forgotten clean up. Is there a reason why this is not solved this way?

**Comment by martindeegan** [comment]
Yeah, that would be preferable. The crate is very bare bones right now because I created it for another project I was working on and I just needed a few functions from the library, I haven't had time to update it much lately because of school. If you make a pull request, I'll be happy to look at it. I might work on it a bit this weekend.
I talked with Lolzballs in #3 about making the crate more rust worthy. You can see his method of implementing serial which implements drop.
